rfc/0011-audit-checks.md

```diff
@@ -0,0 +1,354 @@
+
+|              |                                  |
+| :----------- | :------------------------------- |
+| Feature Name | Audit checks                           |
+| Start Date   | 08/05/2022                        |
+| Category     | enhancement,feature                 |
+| RFC PR       | https://github.com/kubewarden/rfc/pull/10  |
+| State        | **IN REVIEW**                     |
+
+
+# Summary
+
+The audit checks inspects the resources already deployed in the cluster and flags
+when something violates some policies installed in the cluster.
+
+# Motivation
+
+Currently Kubewarden can be used to evaluate Kubernetes resources only when a relevant
+operation happens: creation, update, deletion of the resource. However, policies evolve over time.
+More policies can be added or they can be redeployed with different settings.
+As a result of that, a Kubernetes resource that was marked as valid even a couple of hours ago,
+could be instead rejected later on because of a policy change.
+
+We want to implement a audit check that inspects all the resources that are already
+present inside of the cluster and flags all the ones that are violating at least one of
+the currently enforced policies.
+
+## Examples / User Stories
+
+As a operator, I want to know if some resource in my cluster violates the latest policies.
+
+As a user, I want to know if some of resource that I own violates the cluster compliance policies.
+
+As a Kubernetes developer, I want access to the audit checks results to allow me pro-grammatically act upon it.
+
+# Detailed design
+
+The audit check will run at specified intervals and will produce audit reports using the `PolicyReport` format.
+There is going to be one `PolicyReport` object per namespace, plus one `ClusterPolicyReport` that will hold information
+about the cluster-wide resources.
+
+Details about the `PolicyReport` format can be found inside of this [RFC](https://github.com/kubewarden/rfc/pull/13/files).
+
+## Reducing the scope to keep things simple
+
+In the beginning we should reduce the scope of the audit checks to be able to
+ship on time and iterate over the implementation by using real world data.
+
+### No audit for policies targeting `*`
```

Comment:
I would prefer these paragraph to be about all policies that are not auditable.
We should also mention policies that look at the user of the request.

At the end, are we adding an opt-in spec field and metadata field to (Cluster)AdmissionPolicies  that marks them as auditable?